web/satellite/src/components/common/VDatePicker.vue

```diff
@@ -0,0 +1,561 @@
+						<div class="week">
+							<ul>
+								<li v-for="week in weekLibrary" :key="week">{{week}}</li>
+							</ul>
+						</div>
+						<div class="day" v-for="(day, index) in dayList" :key="index" @click="checkDay(day)" :class="{'checked':day.checked,'unavailable':day.unavailable,'passive-day': !(day.inMonth), 'today': day.today}" :style="day.checked ? (option.color && option.color.checkedDay ? { background: option.color.checkedDay } : { background: '#2683FF' }) : {}">{{day.value}}</div>
+					</div>
+				</div>
+				<div class="cov-date-box list-box" v-if="isYearChoiceShown">
+					<div class="cov-picker-box date-list" id="yearList">
+						<div class="date-item year" v-for="yearItem in yearLibrary" :key="yearItem" @click="setYear(yearItem)">{{yearItem}}</div>
+					</div>
+				</div>
+				<div class="cov-date-box list-box" v-if="isMonthChoiceShown">
+					<div class="cov-picker-box date-list">
+						<div class="date-item month" v-for="monthItem in monthLibrary" :key="monthItem" @click="setMonth(monthItem)">{{monthItem}}</div>
+					</div>
+				</div>
+			</div>
+		</div>
+	</div>
+</template>
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+import { DateGenerator, DateStamp, DayAction, DayItem, DisplayedType, Options } from '@/utils/datepicker';
+
+@Component
+    export default class VDatePicker extends Vue {
+
+    @Prop({default: () => new Options()})
+    private option: Options;
+    @Prop({default: () => false})
+    private isSundayFirst: boolean;
+
+    private readonly MAX_DAYS_SELECTED = 2;
+    private showType: number = DisplayedType.Day;
+    private dateGenerator: DateGenerator = new DateGenerator();
+    public isChecking = false;
+
+    public weekLibrary: string[] = [];
+    public monthLibrary: string[] = [];
+    public displayedMonth: string;
+    public selectedDateState: DateStamp = new DateStamp(0, 0, 0);
+    public dayList: DayItem[] = [];
+    public selectedDays: Date[] = [];
+    public yearLibrary: number[] = [];
+
+    public constructor() {
+        super();
+
+        this.weekLibrary = this.isSundayFirst ? this.option.sundayFirstWeek : this.option.mondayFirstWeek;
+        this.monthLibrary = this.option.month;
+        this.displayedMonth = this.monthLibrary[0];
+        this.yearLibrary = this.dateGenerator.populateYears();
+    }
+
+    /**
+	 * computed value that indicates should days view be shown
+	 */
+    public get isDaysChoiceShown(): boolean {
+        return this.showType === DisplayedType.Day;
+    }
+
+    /**
+	 * computed value that indicates should month choice view be shown
+	 */
+    public get isMonthChoiceShown(): boolean {
+        return this.showType === DisplayedType.Month;
+    }
+
+    /**
+	 * computed value that indicates should year choice view be shown
+	 */
+    public get isYearChoiceShown(): boolean {
+        return this.showType === DisplayedType.Year;
+    }
+
+    /**
+	 * nextMonth set previous month
```

Comment:
should it be * onPreviousMonthClick sets previous month? or  * onPreviousMonthClick method sets previous month?